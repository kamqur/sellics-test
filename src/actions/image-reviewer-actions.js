const START_FETCH_IMAGE = "START_FETCH_IMAGE";
const COMPELETE_FETCH_IMAGE = "COMPELETE_FETCH_IMAGE";
const ERROR_FETCH_IMAGE = "ERROR_FETCH_IMAGE";
const ACCEPT_IMAGE = "ACCEPT_IMAGE";
const REJECT_IMAGE = "REJECT_IMAGE";

export {
	START_FETCH_IMAGE,
	COMPELETE_FETCH_IMAGE,
	ERROR_FETCH_IMAGE,
	ACCEPT_IMAGE,
	REJECT_IMAGE
}