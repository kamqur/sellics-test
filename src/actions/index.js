import {
	START_FETCH_IMAGE,
	COMPELETE_FETCH_IMAGE,
	ERROR_FETCH_IMAGE,
	ACCEPT_IMAGE,
	REJECT_IMAGE
} from "./image-reviewer-actions";


export {
	START_FETCH_IMAGE,
	COMPELETE_FETCH_IMAGE,
	ERROR_FETCH_IMAGE,
	ACCEPT_IMAGE,
	REJECT_IMAGE
}