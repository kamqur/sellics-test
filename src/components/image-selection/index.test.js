
import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

configure({adapter: new Adapter()});
import ImageSelection from "./index";

import { ACCEPT_IMAGE, REJECT_IMAGE } from "../../actions";

describe('Image selection test suite', function() {

  const imageURL = "https://upload.wikimedia.org/wikipedia/commons/thumb/b/b6/Image_created_with_a_mobile_phone.png/640px-Image_created_with_a_mobile_phone.png"
  const image = {
    id: "id",
    urls: {
      small: imageURL
    }
  }


  it('Snapshot Testing',() => {
    const fetchRandomImage = jest.fn()
     const component = shallow(<ImageSelection image={image} fetchRandomImage={fetchRandomImage} />)
     expect(component).toMatchSnapshot();
  })


  it('should render image with approiate url', function() {
    const fetchRandomImage = jest.fn()
    const component = shallow(<ImageSelection image={image} fetchRandomImage={fetchRandomImage} />)
    expect(component.find("img").getElement().props.src).toEqual(imageURL)
    expect(component.find("img").getElement().props.alt).toEqual("id")
  });

  it("should accept image", function() {
    const fetchRandomImageMock = jest.fn()
    const component = shallow(<ImageSelection image={image} fetchRandomImage={fetchRandomImageMock} />)
    component.find("#accept").simulate("click");
    expect(fetchRandomImageMock).toBeCalledWith(ACCEPT_IMAGE)
  });

  it("should reject image", function() {
    const fetchRandomImageMock = jest.fn()
    const component = shallow(<ImageSelection image={image} fetchRandomImage={fetchRandomImageMock} />)
    component.find("#reject").simulate("click");
    expect(fetchRandomImageMock).toBeCalledWith(REJECT_IMAGE)
  });

});