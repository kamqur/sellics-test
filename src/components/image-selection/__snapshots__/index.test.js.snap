// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image selection test suite Snapshot Testing 1`] = `ShallowWrapper {}`;
